### DISCUSSION TITLE
Some trouble with encryption

### DISCUSSION THREAD

**Discussion by poperigby** [twpayne/chezmoi#1308] [Q&A]
Hey. I encrypted my `.ssh` folder, but now it prints everything I've encrypted every time I run `chezmoi status`. Is there a way to get this to stop?

**Comment by twpayne** [comment]
Please give details of the exact commands that you ran. There is not enough information in your question to diagnose what is happening.

**Reply by poperigby** [reply]
My bad. 

1. I created a PGP key with my email
2. I put this into my `chezmoi.toml`:
```
[gpg]
    recipient = "user@email.com"
```
3. I ran `chezmoi add --encrypt .ssh`
4. I ran `chezmoi git commit -m "Add .ssh dir"` and then `chezmoi git push`

Now, when I run `chezmoi status`, it prompts me to enter my PGP key passphrase. If I do that, it prints all of this:
```
gpg: encrypted with 2048-bit RSA key, ID 017F8EE68B2C27AE, created 2021-07-15
      "My Username <user@email.com>"
gpg: encrypted with 2048-bit RSA key, ID 017F8EE68B2C27AE, created 2021-07-15
       "My Username <user@email.com>"
gpg: encrypted with 2048-bit RSA key, ID 017F8EE68B2C27AE, created 2021-07-15
      "My Username <user@email.com>"
...
```

**Reply by twpayne** [reply]
I think `gpg.recipient` needs to be the key ID, not an email address. What happens if you put

```toml
[gpg]
    recipient = "017F8EE68B2C27AE"
```

in your config file?

**Reply by poperigby** [reply]
Changing that line doesn't fix it.

**Reply by nandalopes** [reply]
I guess you want to add --quiet to gpg args

```yaml
gpg:
  recipient: "017F8EE68B2C27AE"
  args:
    - --quiet
```

**Reply by poperigby** [reply]
Well, that does get rid of the gpg messages, but it also makes `chezmoi status` hang, which is a little bit frustrating. Why is it informing me of every encrypted file even if those files haven't changed?

**Reply by twpayne** [reply]
Could you provide a sequence of steps that demonstrate the problem?

chezmoi has [a bunch of tests that test gpg encryption](https://github.com/twpayne/chezmoi/blob/master/internal/cmd/testdata/scripts/gpgencryption.txt) (here's the [gpg configuration generator](https://github.com/twpayne/chezmoi/blob/master/internal/cmd/main_test.go#L247-L296)).

**Reply by poperigby** [reply]
The sequence of steps that demonstrate the `chezmoi status` hang? Those steps would just be this...

> 1. I created a PGP key with my email
> 2. I put this into my `chezmoi.toml`:
> ```
> [gpg]
>     recipient = "user@email.com"
> ```
> 3. I ran `chezmoi add --encrypt .ssh`
> 4. I ran `chezmoi git commit -m "Add .ssh dir"` and then `chezmoi git push`

...plus me adding `args = ["--quiet"]` under the `[gpg]` section in my `chezmoi.toml`. It then hangs if I run `chezmoi status`, and then outputs and changed files. I haven't changed any of the .ssh files since I added them.

**Reply by twpayne** [reply]
Please can you provide the exact commands that you ran, not just descriptions of the commands.

Do you need to provide a key ID instead of an email in `gpg.recipient`?

**Reply by poperigby** [reply]
My bad. What commands are you talking about specifically? I thought I did give the full commands.

**Reply by twpayne** [reply]
Could you give the commands you used to create the PGP key? There can be some subtle issues here, e.g. whether the key is marked as ultimately trusted.

**Reply by poperigby** [reply]
Oh, that might be a problem, because I created it with GNOME Keyring. Should I give you the information from `gpg --list-keys`, or should I create a new one?
